### PR TITLE
Update files.py

### PR DESCRIPTION
**Pull request by agtoever**
Refactored entire module. Added docstrings. Added support for opening tsgf files. Improved error handling.

### CODE REVIEW

File: src/files.py [hunk 1]
```diff
@@ -1,75 +1,179 @@
...
+    MOVES_STARTLINE = 13
+    PLAYER_LINES = [3, 4]
+    COMMENT_CHAR = '#'
 
-LINE_OFFSET = 13
+    try:
+        players = [content[linenr].split(COMMENT_CHAR)[0]
+                   for linenr in PLAYER_LINES]
+    except:
+        raise ValueError("Can't read player names from T1 file.")
```
Comment (stevens68) [inline src/files.py:86]: I think we should handle this with a popup withhout reraising (like the case cannot read player names) so the program doesn't crash

File: src/files.py [hunk 2]
```diff
@@ -1,75 +1,179 @@
...
 
+    try:
+        moves = [str2twixt(move) for move in content[MOVES_STARTLINE:]
+                 if len(move) > 0]
+    except:
+        # Just pass on the exception from str2twixt
+        raise
```
Comment (stevens68) [inline src/files.py:93]: dito (popup message something like "unknown error in file....")

File: src/files.py [hunk 3]
```diff
@@ -1,75 +1,179 @@
...
+
+    Examples:
+        >>> content = [['(;FF[4]EV[twixt.ld.DEFAULT]PB[agtoever]PW[Jan Krabbenbos]SZ[24]SO[https://www.littlegolem.net];b[pl];r[ps];b[pr];r[rt];b[ot];r[po];b[pn];r[qq];b[op];r[pg];b[nh];r[oj];b[oi];r[qi];b[nk];r[nf];b[mf])']]
+        >>> parse_tsgf_file(content)
+        (['agtoever', 'Jan Krabbenbos'], [p12, p19, p18, r20, o20, p15, p14, q17, o16, p7, n8, o10, o9, q9, n11, n6, m6])
+    """
+    PLAYERS_STR = ['PB', 'PW']
+    
+    if len(content) > 1:
+        raise ValueError('Found more than 1 line in a tsgf file.')
```
Comment (stevens68) [inline src/files.py:118]: dito

File: src/files.py [hunk 4]
```diff
@@ -1,75 +1,179 @@
...
-                lt.popup(file_name + ": invalid move in line " +
-                         str(i + LINE_OFFSET + 1) + ": " + m)
-                return None
-
-    return moves
+        player_idx = [content[0].find(key) for key in PLAYERS_STR]
+        players = [content[0][idx + 3:content[0].find(']', idx)]
+                   for idx in player_idx]
+    except:
+        raise ValueError("Can't read player names from tsgf file.")
```
Comment (stevens68) [inline src/files.py:125]: dito

File: src/files.py [hunk 5]
```diff
@@ -1,75 +1,179 @@
...
+        
+    try:
+        raw_moves = [line[2:4] for line in content[0].split(';')
+                     if line[:2] == 'r[' or line[:2] == 'b[']
+        moves = list(map(str2twixt, raw_moves))
+    except:
+        # Just pass on the exception from str2twixt
+        raise
```
Comment (stevens68) [inline src/files.py:133]: dito

**Review by stevens68 — CHANGES_REQUESTED** [review]
Hi agtoever
thanks for your contribution!! Well done! I just added one remark (multiple occurences): Would like to keep the program going in case of unreadable files.

I also found a bug (not related to your change): It's not possible to undo the first move of a game. After you try you cannot even reset the board. Will have to look into it.

I meanwhile replaced time.clock in master branch; I think you hit that issue, too.   So did bob440 when trying to run twixtbot-ui on Python 3.8 (ubuntu)

File: src/files.py (hunk 1 above)
Comment (agtoever) [reply src/files.py:86]: Agree. I'll catch and handle it in get_game(). I like to focus user interaction to a restricted number of functions.

File: src/files.py (hunk 2 above)
Comment (agtoever) [reply src/files.py:93]: Agree. I'll catch and handle it in get_game(). I like to focus user interaction to a restricted number of functions.

File: src/files.py (hunk 3 above)
Comment (agtoever) [reply src/files.py:118]: Agree. I'll catch and handle it in get_game(). I like to focus user interaction to a restricted number of functions.

File: src/files.py (hunk 5 above)
Comment (agtoever) [reply src/files.py:133]: Agree. I'll catch and handle it in get_game(). I like to focus user interaction to a restricted number of functions.

File: src/files.py (hunk 4 above)
Comment (agtoever) [reply src/files.py]: This is exception is already handled (and does not lead to an application crash)

**Comment by agtoever** [comment]
Added patch-2 with error handling in get_game(). 
I like to restrict user interaction (such as popups) to a restricted number of functions.
This makes the functions without user interaction much easier to test and maintain.

**Review by stevens68 — APPROVED** [review]
thanks for the changes!

**Pull request merged**